app/assets/javascripts/Components/Helpers/chart_helpers.js

```diff
@@ -2,18 +2,34 @@ import * as I18n from "i18n-js";
 
 export const chartScales = {
   x: {
+    grid: {
+      borderColor: "#98c7ff", // --primary_one
```

Comment:
@therealfake I looked into this a little bit more. The `document.documentElement.style.getPropertyValue("--primary_one")` is correct; the issue is that these values are set using the "light" theme as default, and then mutated in the `set_theme` function when the page loads. But because `chartScales` is a constant, it is initialized using the initial values.

To fix this, you can turn `chartScales` into a *function* that returns an object; then whenever `chartScales` is called, it'll return the colours based on the current theme values. Note that you'll need to update _everywhere_ `chartScales` is used, not just the assignment chart.

Reply:
Oh, interesting. I converted chartScales to a function and all graphs are coherently styled in dark mode now.